### PR TITLE
build: add rules for checking header file, golangci-lint and go modules

### PR DESCRIPTION
**Pull request by cpanato**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Add rules to run some checks similar to other repositories

Test infra PR to add prow jobs: https://github.com/kubernetes/test-infra/pull/20414

#### Which issue(s) this PR fixes:

None

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

/assign @justaugustus @saschagrunert

### CODE REVIEW

File: hack/verify-boilerplate.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+VERSION=v0.1.1
```
Comment (justaugustus) [inline hack/verify-boilerplate.sh:21]:
```suggestion
VERSION=v0.2.0
```

**Review by justaugustus — CHANGES_REQUESTED** [review]
One nit!

File: hack/verify-boilerplate.sh (hunk 1 above)
Comment (cpanato) [reply hack/verify-boilerplate.sh]: done. PTAL

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Pull request merged**